src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java

```diff
@@ -448,6 +450,56 @@ public void onTick() {
         entityLabel.setFontRenderer(fontRenderer);
         frames.add(frame);
 
+        /*
+         * Potion Effects
+         */
+        frame = new Frame(getTheme(), new Stretcherlayout(1), "Potions");
+        frame.setCloseable(false);
+        frame.setPinnable(true);
+        Label potionsLabel = new Label("");
+        potionsLabel.addTickListener(new TickListener() {
+            Minecraft mc = Minecraft.getMinecraft();
+
+            @Override
+            public void onTick() {
+                potionsLabel.setText("");
+
+                if(mc.player == null){
+                    return;
+                }
+
+                // loop through potions and add them to the potions label
+                mc.player.getActivePotionMap().forEach((potion, potionEffect) -> {
+                    String name = I18n.format(potion.getName());
+                    String amplifier;
+                    switch (potionEffect.getAmplifier()) {
+                        case 0:
+                            amplifier = "I";
+                            break;
+                        case 1:
+                            amplifier = "II";
+                            break;
+                        default:
+                            amplifier = "?";
```

Comment:
this should support up to X I think.

Reply:
Minecraft 1.12 code supports up to V
```
String s1 = I18n.format(potion.getName());

                if (potioneffect.getAmplifier() == 1)
                {
                    s1 = s1 + " " + I18n.format("enchantment.level.2");
                }
                else if (potioneffect.getAmplifier() == 2)
                {
                    s1 = s1 + " " + I18n.format("enchantment.level.3");
                }
                else if (potioneffect.getAmplifier() == 3)
                {
                    s1 = s1 + " " + I18n.format("enchantment.level.4");
                }
``` 
I can add up to X if you want but its gonna be an ugly switch statement unless we want to add something from stackoverflow. https://stackoverflow.com/a/19759564/13086246

Reply:
Oh I was just going to make a Kotlin util. It's not that ugly in Kotlin

Reply:
psudocode

```kotlin
return when (level) {
    0 -> "I"
    1 -> "II"
    2 -> "III"
    3 -> IV"
}
```
etc

Reply:
Thats pretty hot ngl. Should I add it to a specific utils class or just make one specifically for that function, or leave it to you for that?

Reply:
I'll do it myself layer :+1: